libraries/Microsoft.Bot.Builder.Runtime/Extensions/IServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Microsoft.Bot.Builder.Dialogs.Adaptive.Conditions;
+using Microsoft.Bot.Builder.Dialogs.Declarative.Resources;
+using Microsoft.Bot.Builder.Runtime.Providers;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.Bot.Builder.Runtime.Extensions
+{
+    public static class IServiceCollectionExtensions
+    {
+        public static void AddBotCore(this IServiceCollection services, IConfiguration configuration)
+        {
+            if (services == null)
+            {
+                throw new ArgumentNullException(nameof(services));
+            }
+
+            if (configuration == null)
+            {
+                throw new ArgumentNullException(nameof(configuration));
+            }
+
+            // Component registrations must be added before the resource explorer is instantiated to ensure
+            // that all types are correctly registered. Any types that are registered after the resource explorer
+            // is instantiated will not be picked up otherwise.
+
+            ComponentRegistrations.Add();
+
+            string applicationRoot = configuration.GetSection(ConfigurationConstants.ApplicationRootKey).Value;
+
+            services.AddBotCore(
+                configuration,
+                resourceExplorerImplementationFactory: (serviceProvider) =>
+                    new ResourceExplorer()
+                        .AddFolder(applicationRoot)
+                        .RegisterType<OnQnAMatch>(OnQnAMatch.Kind));
+        }
+
+        internal static void AddBotCore(
```

Comment:
This adds the corebot runtime configuration. is that correct? I'm not seeing how this is specific to corebot.

Reply:
I've renamed this - this is really about applying the standard runtime configuration and isn't really related to CoreBot (the class or the template). I've renamed to AddBotRuntime instead. Again, this will come in subsequent branch, so will leave this open for now.

Reply:
Opened issue.